src/averageAges.js

```diff
@@ -15,11 +15,25 @@
  * @return {number}
  */
 function calculateMenAverageAge(people, century) {
-  // write code here
-  // learn how to use array methods like .filter .map .some .every .find .reduce
-  // avoid using loop and forEach
-  // replace `if ()` statement with &&, || or ?:
-  // without nesting
+  return people
+    .filter(({ sex, died }) => {
+      const deathCentury = Math.ceil(died / 100);
+      const isCenturyRequired = century
+        ? sex === 'm' && deathCentury === century
+        : sex === 'm';
+
+      return isCenturyRequired;
+    })
+    .reduce((acc, person, index, array) => {
+      const { born, died } = person;
+      const age = died - born;
+
+      if (index === array.length - 1) {
```

Comment:
1. `array `- it is better not to give such general names.
2. Since our array does not change, we can access it directly without accepting an extra value from the arguments: `people.length`

Reply:
done